### PR TITLE
Cleanup program docs

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Programs aren't in the docs.

#### Summary of Changes

* Working in the direction of generated docs by improving markdown, consistency and reducing duplication.
* Add signers to the list of account references

### CODE REVIEW

File: programs/stake/src/stake_instruction.rs [hunk 1]
```diff
@@ -43,86 +43,75 @@ impl<E> DecodeError<E> for StakeError {
...
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub enum StakeInstruction {
-    /// `Initialize` a stake with Lockup and Authorized information
+    /// Initialize a stake with lockup and authorization information
     ///
-    /// Expects 2 Accounts:
-    ///    0 - Uninitialized StakeAccount
-    ///    1 - Rent sysvar
+    /// # Account references
+    ///   0. [WRITE] Uninitialized stake account
```
Comment (jackcmay) [inline programs/stake/src/stake_instruction.rs:49]: I like the effort to formalize these instruction definitions.  How about in addition to WRITE and READ we also add SIGNER.  Also, can we assume `READ` and just add `[WRITE]` or `[WRITE, SIGNER]` as appropriate?

File: sdk/src/loader_instruction.rs [hunk 2]
```diff
@@ -8,23 +8,22 @@ use crate::{
 pub enum LoaderInstruction {
     /// Write program data into an account
     ///
-    /// * key[0] - the account to write into.
-    ///
-    /// The transaction must be signed by key[0]
+    /// # Account accesses
+    ///   * 0. [WRITE, SIGNER] Account to write to
     Write {
         offset: u32,
```
Comment (jackcmay) [inline sdk/src/loader_instruction.rs:14]: Should we also add comments for these members?

File: programs/stake/src/stake_instruction.rs (hunk 1 above)
Comment (garious) [reply programs/stake/src/stake_instruction.rs:49]: Yes, I'm running under the assumption that WRITE implies READ.

Comment (jackcmay) [reply programs/stake/src/stake_instruction.rs:49]: Yeah, can we get rid of the ubiquitous `READ`?

Comment (garious) [reply programs/stake/src/stake_instruction.rs:49]: Hmm, I see.  But that leaves us with a bunch of docs that say `0. [] Some read-only account`.

Comment (garious) [reply programs/stake/src/stake_instruction.rs:49]: Here's what it looks like 28d922d

File: sdk/src/loader_instruction.rs (hunk 2 above)
Comment (garious) [reply sdk/src/loader_instruction.rs:15]: Done

**Pull request merged**